### ISSUE TITLE
Can't comprehend how to add "localImage" via JSON under Images.main 

### ISSUE THREAD

**Issue by anantthebiker** [spotify/HubFramework#148]
When you look @ this doc: https://spotify.github.io/HubFramework/Protocols/HUBComponentImageData.html 
it tells we can add local Images from the swift code via "localImage" property of HUBComponentImageData.
Whereas in the JSON Doc. : https://spotify.github.io/HubFramework/json-programming-guide.html 
There is only a reference to URI & placeholder property under Images.main.

<img width="879" alt="screen shot 2016-11-11 at 12 05 42 pm" src="https://cloud.githubusercontent.com/assets/8290135/20206548/39f608b2-a807-11e6-8939-d0476daa45c2.png">

So how do I give reference to LocallyStored Images from the JSON file, via which attribute? 
May be I am missing something or the doc misses something. 
Thanks
Anant

**Comment by JohnSundell** [comment]
@anantthebiker The framework doesn't support adding local images when using JSON yet, it needs to be done in code in a content operation. We could always add support for specifying a local image name in JSON as well (we haven't had the need for it internally yet, since when we use JSON we always use remote images downloaded over the network).

@diegc, do you think this is something that would be OK to add to the JSON schema from an Android perspective?

**Comment by JohnSundell** [comment]
How about a `local` key that points to a bundled image?

**Comment by diegc** [comment]
@JohnSundell there would be no need to add a new property to the schema from the Android perspective, since loading local images can be done using the standard uri format for local resources (`android:resource://...`). The code path for local resources is the same as for remote resources.

**Comment by JohnSundell** [comment]
@diegc Ah that's cool! But if one wants to use the same JSON for both iOS & Android, `android:resource//...` cannot be used (as it won't work on iOS). How about we introduce a `local` key that then gets translated to the platform-specific paths on the client side?

**Comment by diegc** [comment]
@JohnSundell definitely. I was just referring to the ability to set a local image in code on Android. While the same property can be used for both local and remote if the clients inspect its value, I agree that for cross-platform JSON, having something more explicit makes it easier (Android could then, for instance, translate `local = "foo_icon"` to something like `uri = "android:resource//org.blah/drawable/foo_icon"`).

So, in short, adding `local` is ok for Android :)

**Comment by JohnSundell** [comment]
@diegc Sounds awesome! 🚀 I'll add an issue for adding it.

**Comment by JohnSundell** [comment]
Implementation plan here: https://github.com/spotify/HubFramework/issues/162
